### PR TITLE
Free space actually deletes items

### PR DESCRIPTION
**Pull request by lfdversluis**
This PR fixes the bug where the freespace method does not delete anything. 

Fixes **1** bug
Added **1** (property) method for lint issues (enhancement). This can be used by other classes also accessing the protected member `_connection`.
Adds **2**  and modifies **1** tests.

### CODE REVIEW

**Comment by lfdversluis** [comment]
"Tests on Mac — An error occurred during the tests on Linux. " ?

**Comment by lfdversluis** [comment]
Build was aborted
Aborted by anonymous

**Comment by lfdversluis** [comment]
retest this please

**Comment by lfdversluis** [comment]
@whirm Why are the tests behaving so weird on this PR?

**Comment by devos50** [comment]
@lfdversluis I just figured out that the tests on Mac are failing (https://jenkins.tribler.org/job/GH_Tribler_PR_tests_mac/16/testReport/junit/Tribler.Test.Core.test_sqlitecachedbhandler/TestTorrentDBHandler/test_freeSpace/) which cause the Tests on Linux job to abort immediately and report a failure.

**Comment by lfdversluis** [comment]
@devos50 wow, so that message "an error occured during the tests on linux" wasn't copy pasta. Thanks for letting me know.

**Comment by lfdversluis** [comment]
@devos50 Thinking about it.. it makes even less sense because the current code should also break then (update vs delete on the same table).

**Comment by NielsZeilemaker** [comment]
It's not the same table, torrent vs torrenttracker

**Comment by lfdversluis** [comment]
@NielsZeilemaker Ah yes, my bad. Do you remember why this function was disabled in the past?

**Comment by NielsZeilemaker** [comment]
Deleting from the database doesn't make a lot of sense, you want to get rid of the torrent file. Also looking at the comment below, keeping the foreign key available for other tables makes sense. 
Especially the channels have a link to the torrentid

**Comment by lfdversluis** [comment]
@NielsZeilemaker I am not too familiar with the current logic. Would deleting the file in `torrent_file_name` (in the Torrent table), lead to the deletion of the correct .torrent file?

**Comment by NielsZeilemaker** [comment]
Not sure, but it would cause the torrent to not show up in the collectedtorrent view,  and hence it wouldn't be send to others etc. 
The newish torrent database has been created after I left

**Comment by lfdversluis** [comment]
@NielsZeilemaker Scratch that last, I just found an upgrade "replace torrent_file_name column with is_collected column" in `_upgrade_25_to_26`. Looks like solving this requires knowledge of the details. This will take some time I guess..

**Comment by whirm** [comment]
The torrent store is a key values store, the key is the infohash an the value the raw contents of the .torrent file.

If is_collected is true, it means we've got the torrent on the store, if it's false it means it's not. So if you set is_collected to false when cleaning, you first need to delete that entry from the torrent store and a quick grep through Tribler doesn't show a single place where the torrents are removed from the store.

I also think we could store way more than 50.000 torrents in levelDB if that improves the user experience.

**Comment by devos50** [comment]
You can easily delete a torrent from the leveldb torrent store by calling `del session.lm.torrent_store[infohash]`. However, maybe it is better to create a separate method in `Session` (called `delete_collected_torrent` or something similar) and use that.

I also think we can safely increase the limit of collected torrents to 100K or even more. [This](http://highscalability.com/blog/2011/8/10/leveldb-fast-and-lightweight-keyvalue-database-from-the-auth.html) post talks about scalable leveldb databases with more than a million entries:

- _Search performance is still O(lg N) with a very large branching factor (so the constant factor is small and number of seeks should be <= 10 even for gigantic databases).
- One early user found performance degraded at around 200 million keys._

Maybe @lfdversluis can do a performance measurement regarding leveldb? Should be easy to setup, we can just fill a database with many torrent files and perform queries on it :)

**Comment by lfdversluis** [comment]
@devos50 I wouldn't do too many benchmarks now. I would first focus on issues and then on improvements including benchmarks to show it's indeed improving.

**Comment by NielsZeilemaker** [comment]
@devos50 what do you want to achieve by increasing the limit to 100k?

**Comment by lfdversluis** [comment]
The torrent store is enabled in the tests, yet they report it is disabled somehow. Any idea what could be wrong @devos50 @whirm ?

**Comment by lfdversluis** [comment]
retest this please

**Comment by lfdversluis** [comment]
@whirm Ready to be reviewed.

File: Tribler/Core/Session.py [hunk 1]
```diff
@@ -601,6 +601,20 @@ def save_collected_torrent(self, infohash, data):
             raise OperationNotEnabledByConfigurationException("torrent_store is not enabled")
         self.lm.torrent_store.put(hexlify(infohash), data)
 
+    def delete_collected_torrent(self, infohash):
+        """
+        Deletes the given torrent into the torrent_store database.
```
Comment (whirm) [inline Tribler/Core/Session.py]: s/into/from/

File: Tribler/Core/Session.py [hunk 2]
```diff
@@ -601,6 +601,20 @@ def save_collected_torrent(self, infohash, data):
...
+        :param infohash: The given infohash binary.
+        """
+        if not self.get_torrent_store():
+            raise OperationNotEnabledByConfigurationException("torrent_store is not enabled")
```
Comment (whirm) [inline Tribler/Core/Session.py]: Try to make lines shorter than 120

Comment (whirm) [inline Tribler/Core/Session.py]: Not sure why pylint is not catching this. Did I forget to configure this?

File: Tribler/Core/Session.py [hunk 3]
```diff
@@ -601,6 +601,20 @@ def save_collected_torrent(self, infohash, data):
...
+
+        if self.lm.torrent_store is not None:
+            del self.lm.torrent_store[hexlify(infohash)]
+        else:
+            self._logger.error("Torrent store is enabled but could not delete infohash %s" % infohash)
+            raise RuntimeError("Torrent store is enabled but None")
```
Comment (whirm) [inline Tribler/Core/Session.py]: No need for this kind of checks, it will fail anyway. just leave the del statement on this chunk.

**Comment by lfdversluis** [comment]
@whirm looks like it

File: Tribler/Test/Core/test_session.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        config.set_torrent_store(False)
+        session = Session(config, ignore_singleton=True)
+        session.delete_collected_torrent(None)
+
+    @raises(RuntimeError)
+    def test_torrent_store_none(self):
+        config = SessionStartupConfig()
+        config.set_torrent_store(True)
+        session = Session(config, ignore_singleton=True)
+        session.delete_collected_torrent(None)
```
Comment (whirm) [inline Tribler/Test/Core/test_session.py]: Missing EOL

Comment (whirm) [inline Tribler/Test/Core/test_session.py]: This should be caught by the test job too

File: Tribler/Test/Core/test_sqlitecachedbhandler_torrents.py [hunk 5]
```diff
@@ -224,10 +230,14 @@ def test_getCollectedTorrentHashes(self):
 
     @blocking_call_on_reactor_thread
     def test_freeSpace(self):
+        # Manually set the torrent store because register is not called.
+        self.session.lm.torrent_store = LevelDbStore(self.session.get_torrent_store_dir())
+        # print self.session.lm.torrent_store
```
Comment (whirm) [inline Tribler/Test/Core/test_sqlitecachedbhandler_torrents.py]: leftover debug stuff

File: Tribler/Test/Core/test_sqlitecachedbhandler_torrents.py [hunk 6]
```diff
@@ -224,10 +230,14 @@ def test_getCollectedTorrentHashes(self):
...
         old_res = self.tdb.getNumberCollectedTorrents()
         self.tdb.freeSpace(20)
         res = self.tdb.getNumberCollectedTorrents()
-        self.assertEqual(old_res, res)
+        self.session.lm.torrent_store.close()
+        self.assertLess(res, old_res)
```
Comment (whirm) [inline Tribler/Test/Core/test_sqlitecachedbhandler_torrents.py]: Maybe you should check that the amount left is the right one, not just less than before.

**Comment by whirm** [comment]
Looks fine

File: Tribler/Core/Session.py (hunk 2 above)
Comment (lfdversluis) [inline Tribler/Core/Session.py]: The line is less than 120 according to pycharm.

**Comment by lfdversluis** [comment]
@whirm Ready to merge.

**Comment by lfdversluis** [comment]
retest this please

**Comment by whirm** [comment]
retest this please

**Comment by lfdversluis** [comment]
`TestLibtorrentDownload.test_playdownload - streaming did not start - Condition was not satisfied in 60 seconds`

**Comment by lfdversluis** [comment]
retest this please

**Comment by lfdversluis** [comment]
`TestLibtorrentDownload.test_downloadfromurl - no download progress - Condition was not satisfied in 30 seconds`

**Comment by lfdversluis** [comment]
retest this please

**Comment by lfdversluis** [comment]
retest this please

**Comment by lfdversluis** [comment]
@whirm Ready for review.

**Comment by whirm** [comment]
Looks good, squash the last commit with the one that introduced what you are fixing (get used to do fixup commits) and I'll merge.

**Comment by lfdversluis** [comment]
@whirm Squashed. Tests on mac are failing because something is up with the tester on mac. (see other PRs).

**Comment by lfdversluis** [comment]
retest this please

**Pull request merged**